dts/arm/atmel/samd20.dtsi

```diff
@@ -30,6 +30,7 @@
 
 		tc6: tc@42003800 {
 			compatible = "atmel,sam0-tc32";
+			status = "disabled";
```

Comment:
I wonder if this should be pulled out into its own PR.  Is it the case that some SoC's don't have TC6 at all?  If so, the node shouldn't exist on those SoCs (rather than using 'disabled').

Reply:
The issue is we don't (currently) have granularity below "SAM D20", while the configurations of the chip vary between "SAM D20 J/G/E".  When it's present on a SAM D20 chip it shared the same properties (i.e. the hardware definition is common between all SAM D20, but various configurations may have it or not).  So in that way it makes sense to put it in SAM D20 top level, at least.  

SERCOM has the same issue, but it doesn't come up because they're already "disabled" by the compatible flagging.  That is, they require the overlay to set the compatible property before any defines are generated.

It's actually a bit worse than that if you get into SAM D21, since they have further configuration splitting (e.g. SAMD21E16C vs SAMD21E16L vs SAMD21G16L all have different availability between TCs and SERCOM).  So I mostly just didn't want to get into that.

So if the SAMD2x DTSI files are ever refactored, that would likely be something to incorporate.  Here I just did this because it was the simplest way to make sure the unavailable HAL constants weren't being referenced by default.  With that said, I'm not against removing the node entirely until such a refactor is done, either.